docs/development/new-cloud-provider-botanist.md

```diff
@@ -0,0 +1,61 @@
+    2. Populate the chart with the `yaml` files necessary to use CSI for your cloud provider. See existing examples. You must provide the _entire_ CSI uplift, including the common attacher and provisioner, as well as your cloud provider's specific plugin.
+    3. Add a section for `csi-<provider>` with the images and credentials, and `enabled: false`, to [charts/shoot-core/values.yaml](../../charts/shoot-core/values.yaml). See existing examples.
+6. Add your cloud provider and supported versions to [docs/usage/supported_k8s_versions.md](../../docs/usage/supported_k8s_versions.md)
+7. Add examples to [example/](../example/):
+    1. Create a file `30-cloudprofile-<provider>.yaml`. See existing examples.
+    2. Create a file `50-seed-<provider>.yaml`. See existing examples.
+
+### Golang code
+
+1. If necessary, add support for your cloud provider libraries to [Gopkg.toml](../../Gopkg.toml). 
+2. Add a `case` for your cloud provider as `case gardenv1beta1.CloudProvider<Provider>` to `GetShootCloudProviderWorkers()` in [pkg/apis/garden/v1beta1/helper/helpers.go](../../pkg/apis/garden/v1beta1/helper/helpers.go)
+3. If your cloud provider SDK uses a persistent client object, Create a client implementation for the provider at [pkg/client/](../../pkg/client/) named `pkg/client/<provider>/`. This client will only be used by code that you will implement in the following steps, so it can follow any convention you want. However, by normal convention, it uses the following standard:
+    * Main entry file `client.go`
+    * Any types defined in `types.go`
+4. Add a `case` for your cloud provider as `case gardenv1beta1.CloudProvider<Provider>` to `Maintain()` in [pkg/controller/shoot/shoot\_maintenance\_control.go](../../pkg/controller/shoot/shoot_maintenance_control.go)
+5. Create a cloud botanist implementation for the cloud provider at [pkg/operation/cloudbotanist/](../../pkg/operation/cloudbotanist/) named `<provider>botanist/`. The botanist will be used for primary interfacing with the cloud provider. It should follow these conventions:
+    * The primary entrypoint to create a new instance is `<provider>botanist.New(o *operation.Operation, purpose string)`, and will be called by `cloudbotanist.go`
+    * `New()` is expected to return a `struct` that implements the `CloudBotanist interface` defined in [pkg/operation/cloudbotanist/types.go](../../pkg/operation/cloudbotanist/types.go)
+6. Consume the cloud botanist implementation you created in the previous step by adding a `case gardenv1beta1.CloudProvider<Provider>` to the `func New()` in [pkg/operation/cloudbotanist/cloudbotanist.go](../../pkg/operation/cloudbotanist/cloudbotanist.go)
+7. Define a string `const` for any specific constants you need. These generally are consumed both inside your specific botanist and in the [botanist health check](../../pkg/operation/botanist/health_check.go)
```

Comment:
I don't understand. Can you explain?

Reply:
It was there in an earlier version. I will remove it.